File: src/main/resources/templates/swift/solution/Sources/${packageNameFolder}Lib/BubbleSort.swift
import Foundation

public class BubbleSort: SortStrategy {
    /**
     Sorts dates with BubbleSort.

     - Parameter input: the List of Dates to be sorted
    */
    public func performSort(_ input: inout [Date]) {
        for i in (0 ..< input.count).reversed() {
            for j in 0 ..< i where input[j] > input[j + 1] {
                    let temp: Date! = input[j]
                    input[j] = input[j + 1]
                    input[j + 1] = temp
            }
        }
    }
}

File: src/main/resources/templates/c_sharp/solution/BubbleSort.cs
namespace assignment;

public class BubbleSort : ISortStrategy
{
    /// <summary>
    /// Sorts dates with BubbleSort.
    /// </summary>
    /// <param name="dates">the List of Dates to be sorted</param>
    public void PerformSort(List<DateTime> dates)
    {
        for (int i = dates.Count - 1; i >= 0; i--)
        {
            for (int j = 0; j < i; j++)
            {
                if (dates[j] > dates[j + 1])
                {
                    (dates[j + 1], dates[j]) = (dates[j], dates[j + 1]);
                }
            }
        }
    }
}